crates/nu-command/src/filters/last.rs

```diff
@@ -75,18 +77,29 @@ impl Command for Last {
         let span = call.head;
 
         let rows: Option<i64> = call.opt(engine_state, stack, 0)?;
-        let v: Vec<_> = input.into_iter().collect();
-        let vlen: i64 = v.len() as i64;
-        let beginning_rows_to_skip = rows_to_skip(vlen, rows);
+        let to_keep = rows.unwrap_or(1) as usize;
 
-        if rows.is_some() {
-            let iter = v.into_iter().skip(beginning_rows_to_skip as usize);
+        // early exit for `last 0`
+        if to_keep == 0 {
+            return Ok(PipelineData::new(span).set_metadata(metadata));
+        }
```

Comment:
Question at the general indexing command refactor crew: Should this possibly be an error?

Reply:
I see no reason to error at `to_keep == 0`, an empty list is a valid construct and this is consistent with other explicit arguments. After some thought, I believe this should be
```rust
if to_keep == 0 {
    return Ok(Vec::<Value>::new()
        .into_pipeline_data(engine_state.ctrlc.clone())
        .set_metadata(metadata));
}
```

What do you think about this?